testbed/datasenders/fluent.go

```diff
@@ -0,0 +1,160 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasenders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+const (
+	fluentSplunkHECTestcseDatafileVar = "FLUENT_SPLUNKHEC_TESTCASE_DATA_FILE"
+	fluentPortVar                     = "FLUENT_SPLUNKHEC_FLUENT_PORT"
+)
+
+// FluentLogsForwarder forwards logs to fluent forwader
+type FluentLogsForwarder struct {
+	testbed.DataSenderBase
+	fluentLogger *fluent.Fluent
+	dataFile     *os.File
+}
+
+// Ensure FluentLogsForwarder implements LogDataSender.
+var _ testbed.LogDataSender = (*FluentLogsForwarder)(nil)
+
+func NewFluentLogsForwarder(t *testing.T, port int) *FluentLogsForwarder {
+	var err error
+	portOverride := os.Getenv(fluentPortVar)
+	if portOverride != "" {
+		port, err = strconv.Atoi(portOverride)
+		require.NoError(t, err)
+	}
+
+	f := &FluentLogsForwarder{DataSenderBase: testbed.DataSenderBase{Port: port}}
+
+	if dataFileName := os.Getenv(fluentSplunkHECTestcseDatafileVar); dataFileName != "" {
```

Comment:
Add comments to explain what this if/else cases do.

Reply:
Added a comment with details.